dali/python/nvidia/dali/data_node.py

```diff
@@ -122,6 +139,63 @@ def __bool__(self):
                 " represent computations in DALI Pipeline see the \"Mathematical Expressions\""
                 " section of DALI documentation."))
 
+    def __getitem__(self, val):
+        idxs = []
+        new_axes = []
+        new_axis_names = []
+
+        # returns True if this index adds a new output dimension
+        def process_index(idx, dim):
+            if idx is None:
+                idxs.append((None, None, None, None))
+                return True
+            elif isinstance(idx, slice):
+                if idx.step is not None and idx.step != 1:
+                    raise NotImplementedError("Slicing with non-unit step is not implemented.")
+                idxs.append((None, idx.start, idx.stop, None))
+                return True
+            elif isinstance(idx, _NewAxis):
+                new_axes.append(dim)
+                if idx.name is not None:
+                    new_axis_names.append(idx.name)
+                return True
+            else:
```

Comment:
I think this should be case for integers, and we also can handle `...` (Elipsis).

Reply:
We don't handle ellipsis... yet. It would actually be quite hard to add.

Reply:
ack, that is just me starting with this file, didn't think about the use of ArgumentInputs.